pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java

```diff
@@ -84,14 +85,7 @@ public String getData(
 
     ZNRecord znRecord = _pinotHelixResourceManager.readZKData(path);
     if (znRecord != null) {
-      byte[] serializeBytes = _znRecordSerializer.serialize(znRecord);
-      if (GZipCompressionUtil.isCompressed(serializeBytes)) {
-        try {
-          serializeBytes = GZipCompressionUtil.uncompress(new ByteArrayInputStream(serializeBytes));
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-      }
+      byte[] serializeBytes = serialize(znRecord);
```

Comment:
Not introduced in this PR, but we should be able to directly use `JsonUtils.objectToPrettyString(znRecord)` instead of using the `ZNRecordSerializer`, compressing and then decompressing again

Reply:
it doesn't work because the current way controller UI isn't conform with this. I would suggest let's address this in a separate PR